waku/factory/waku.nim

```diff
@@ -99,9 +109,40 @@ proc validateShards(conf: WakuNodeConf): Result[void, string] =
 
   return ok()
 
+proc setupSwitchServices(
+    waku: Waku, conf: WakuNodeConf, circuitRelay: Relay, rng: ref HmacDrbgContext
+) =
+  proc onReservation(addresses: seq[MultiAddress]) {.gcsafe, raises: [].} =
+    debug "circuit relay handler new reserve event",
+      addrs_before = $(waku.node.announcedAddresses), addrs = $addresses
+
+    waku.node.announcedAddresses.setLen(0) ## remove previous addresses
+    waku.node.announcedAddresses.add(addresses)
+    debug "waku node announced addresses updated",
+      announcedAddresses = waku.node.announcedAddresses
+
+    waku.wakuDiscv5.updateAnnouncedMultiAddress(addresses).isOkOr:
+      error "failed to update announced multiaddress", error = $error
+
+  let autonatService = getAutonatService(rng)
+  if conf.isRelayClient:
+    ## The node is considered to be behind a NAT or firewall and then it
+    ## should struggle to be reachable and establish connections to other nodes
+    const MaxNumRelayServers = 2
+    let autoRelayService = AutoRelayService.new(
+      MaxNumRelayServers, RelayClient(circuitRelay), onReservation, rng
+    )
+    let holePunchService = HPService.new(autonatService, autoRelayService)
+    waku.node.switch.services = @[Service(holePunchService)]
+  else:
+    waku.node.switch.services = @[Service(autonatService)]
+
+    if conf.isRelayClient:
+      let relay = RelayClient.new()
```

Comment:
isn't this part not reachable? because it's in an `else` corresponding to `if conf.isRelayClient:` I think

Reply:
Great catch! thanks for it!